soc/nordic/nrf54h/mram.c

```diff
@@ -5,54 +5,55 @@
  */
 
 #include <zephyr/init.h>
-
-#define MODULE mram_suspend_off
+#include <zephyr/kernel.h>
 #include <zephyr/logging/log.h>
-LOG_MODULE_REGISTER(MODULE);
+#include <zephyr/toolchain.h>
 
 #include <services/nrfs_mram.h>
 #include <nrfs_backend_ipc_service.h>
 
-#define MRAM_SUSPEND_OFF_INIT_PRIO 90
+LOG_MODULE_REGISTER(mram, CONFIG_SOC_LOG_LEVEL);
 
 void mram_latency_handler(nrfs_mram_latency_evt_t const *p_evt, void *context)
 {
+	ARG_UNUSED(context);
+
 	switch (p_evt->type) {
 	case NRFS_MRAM_LATENCY_REQ_APPLIED:
-		LOG_DBG("MRAM latency handler: response received");
+		LOG_DBG("MRAM latency not allowed setting applied");
 		break;
 	case NRFS_MRAM_LATENCY_REQ_REJECTED:
-		LOG_ERR("MRAM latency handler - request rejected!");
-		break;
+		LOG_ERR("MRAM latency not allowed setting rejected");
+		k_panic();
+		__fallthrough;
 	default:
-		LOG_ERR("MRAM latency handler - unexpected event: 0x%x", p_evt->type);
-		break;
+		CODE_UNREACHABLE;
```

Comment:
This could prevent the handler from returning properly if a new event is ever added. I think we should just break here, and maybe keep the "unexpected event" print as a `LOG_DBG`.